### ISSUE TITLE
Building fails:Failed to execute goal com.pyx4me:proguard-maven-plugin:2.0.4:proguard

### ISSUE THREAD

**Issue by 0x6d686b** [rtyley/agit#58]
```
[INFO] proguard jar: /root/.m2/repository/net/sf/proguard/proguard/4.4/proguard-4.4.jar
 [proguard] ProGuard, version 4.4
 [proguard] Reading input...
 [proguard] java.io.IOException: The input doesn't contain any classes. Did you specify the proper '-injars' options?
 [proguard]     at proguard.InputReader.execute(InputReader.java:91)
 [proguard]     at proguard.ProGuard.readInput(ProGuard.java:195)
 [proguard]     at proguard.ProGuard.execute(ProGuard.java:78)
 [proguard]     at proguard.ProGuard.main(ProGuard.java:499)
[INFO]                                                                         
[INFO] ------------------------------------------------------------------------
[INFO] Skipping Agit parent POM project
[INFO] This project has been banned from the build due to previous failures.
[INFO] ------------------------------------------------------------------------
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] Agit parent POM project ........................... SUCCESS [0.001s]
[INFO] agit test utils ................................... SUCCESS [1.481s]
[INFO] Agit - git client for Android ..................... FAILURE [3.131s]
[INFO] Agit integration tests ............................ SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 12.982s
[INFO] Finished at: Fri Nov 25 00:52:02 CET 2011
[INFO] Final Memory: 13M/23M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal com.pyx4me:proguard-maven-plugin:2.0.4:proguard (default) on project agit: Obfuscation failed (result=1) -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <goals> -rf :agit
```

Maven Version 3.0.3
Fresh install on Debian 6 (thus I'm root, on a vm)

**Comment by adpalmer** [comment]
I get the same error on OS X Lion with Maven Version 3.0.3

**Comment by jberkel** [comment]
here's a quick workaround: jberkel/agit@7cdfe7e42050fb228c1b64680b1f759e821f8ac1

**Comment by 0x6d686b** [comment]
WFM

**Comment by rtyley** [comment]
So the build _is_ broken, and the sad thing about that is that it broke with simple commit a014c970 - basically upgrading from android-maven-plugin 3.0.0-alpha-14 to the next released version, which was the final 3.0.0 release.

@jberkel's patch works, in that it will build the main app apk, but it doesn't fix the integration tests. There are proguard and dalvik vm errors that boggle the mind. The fixes for some of these issues seem to be library-version dependant, so I'm taking the opportunity to update Agit to current library versions (particularly of roboguice).

In the mean time, if you want to build Agit, you might want to do it on 068a53b2, the prior commit to a014c970 on the master branch.

**Comment by rtyley** [comment]
The master branch (eg 9c05a83) now compiles without problems, thanks to the upgrade to android-maven-plugin 3 proper and fixed proguard configuration.
